### ISSUE TITLE
Option to set the size of the undo/redo history

### ISSUE THREAD

**Issue by gtournie** [microsoft/vscode#75606]
- VSCode Version: 1.35.1
- OS Version: OSX 10.14.5

Steps to Reproduce:

- Make many changes in a file (I'd say at least 30)
- Try to undo them.

Duplicate of [this issue](https://github.com/microsoft/vscode/issues/10396), which has not been resolved. Undo history is not unlimited as it is suggested.

Would be nice to have an option in the settings to configure the number of undo/redo.

**Comment by V-ed** [comment]
I just tried and was able to undo at least 70 changes, and I could probably go further but got a bit lazy. What I did was simply opening an untitled file, typing alot of text on different lines, then counted the number of times I could undo. In my test, I couldn't go further than 70 because the file got empty at that point.

>Actually there is no limit to the undo history so it must be something else...

In the duplicated issue you linked, @jrieken said that there's no limit, which my 70 undos suggests compared to what you are describing.

It may be one of your extensions that modifies your file when you do your work in a way that deletes your undo history.

From [vscode's Submitting Bugs and Suggestions wiki](https://github.com/Microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions#writing-good-bug-reports-and-feature-requests) :
> Please remember to do the following:
>
>- Search the issue repository to see if there exists a duplicate.
>- **Recreate the issue after disabling all extensions (see the docs for how to do this). If you find the issue is caused by an extension you have installed please file an issue on the extension respectively.**
>- Simplify your code around the issue so we can better isolate the problem.
